### ISSUE TITLE
Possible wiremod expression2 soundPlaying() bug

### ISSUE THREAD

**Issue by raven1934** [wiremod/wire#3091]
Hello, I have tried to make another project which involves checking if a sound is already playing each tick of the chip as it only needs to be put on just once and then the pitch and volume is adjusted during it's use. I however utilized the use of soundPlaying() to check if the sound is already playing to prevent excessive sound resource use, but it seems to be unrealiable at times and causes even second long gaps at the update, this is very bad as during that second my chip reinitiates the sound to start playing and causes massive lag to server and myself... I'm unsure if I can use some conditional myself to prevent this from happening as a bypass, but I wanted to ask if someone more experienced has ran to same problem and if it's a bug or whether there's something I've overlooked. 

The basically simplest setup to replicate the issue in my opinion is just making following setup, as I've managed to in my opinion to narrow it to the sound commands themselves. The result is choppy spammy sound that repeats from the sounds start, and since no other block of commands initiates the soundPlay of that particular sound it has to be here as there's no way the stopSound block is accidentally triggering as it's specific parameters aren't met. Thanks for whoever looks into this stuff btw!

if(!soundPlaying("EntitySound"))
{
Entity:soundPlay("EntitySound",soundDuration("SoundPath"),"SoundPath")
}

runOnTick(1)

**Comment by Fasteroid** [comment]
You *MIGHT* be able to work around soundPlaying not working correctly if you do it yourself with a timer, like this:
```
@persist PlayingSounds:table

function betterSoundPlay(ID:string, Duration:number, Path:string){
    PlayingSounds[ID, number] = 1
    timer("soundEnd_" + ID, Duration * 1000)
}

function number soundIsPlaying(ID:string){
    return PlayingSounds[ID, number] # if it doesn't exist, this will return 0
}

event tick(){
    foreach(ID:string, _:number = PlayingSounds){ # check everything that's playing
        if( clk("soundEnd_" + ID) {
            PlayingSounds:remove(ID)
        }
    }
}
```

Unfortunately, playing E2 sounds on the same id/entity when the previous one is about to end is quite finnicky.

**Comment by raven1934** [comment]
Atm the best workaround is to just set the looping sound to play at start and then adjust the volume to hide the sound in situations it needs to be not on, for sounds that need to start in an event my workaround is to just set it to play for the soundDuration(sound) and then pretend it doesnt exist anymore.
